tee-worker/omni-executor/aa-contracts/src/accounts/OmniAccount.sol

```diff
@@ -102,23 +103,64 @@ contract OmniAccount is BaseAccount, TokenCallbackHandler, UUPSUpgradeable, Init
         override
         returns (uint256 validationData)
     {
-        // UserOpHash can be generated using eth_signTypedData_v4
-        address signer = ECDSA.recover(userOpHash, userOp.signature);
-        if (owner == _determineOa(signer) || isRootSigner(signer)) {
-            return SIG_VALIDATION_SUCCESS;
+        require(userOp.signature.length >= 1, "signature too short");
+
+        UserOpSigType sigType = UserOpSigType(uint8(userOp.signature[0]));
+        bytes calldata sig = userOp.signature[1:];
+
+        if (sigType == UserOpSigType.EOA) {
+            return _validateEOA(userOpHash, sig);
```

Comment:
eoa | root | passkey - they are all same signature types (only ecdsa signature) but checked against different address (owner, one of root keys, one of passkey keys).

Maybe we should have only two signature types ?

Reply:
kk, just a small detail - all 3 (eoa, root, passkey) are **eoa** imho 
so i would rename eoa to owner in order to keep it consistent

Reply:
That's a good point, I'll rename it.

Also does it make more sense to call it `UserOpSignerRole` enum? @kziemianek 
As you mentioned it's same signature type (secp256k1), just different signer roles